packages/theme/src/color-ramps/lib/index.ts

```diff
@@ -68,10 +68,8 @@ function calculateRamp( {
 		value: number;
 	};
 } ) {
-	const rampResults = {} as Record<
-		keyof Ramp,
-		{ color: string; warning: boolean }
-	>;
+	const rampResults = {} as Record< keyof Ramp, string >;
+	let warnings: string[] | undefined;
```

Comment:
`warnings` could always be an array, mostly empty. Adding the `undefined` option just forces users to do extra `?.` checks.

Reply:
The idea with making it `undefined` is that it could be excluded from serialized output, further reducing size (see `default-ramps.ts`). But it's a good point that this benefit is partially offset by having to do all the subsequent optional handling in the code.